src/matcha_ml/core/core.py

```diff
@@ -90,3 +97,80 @@ def remove_state_lock() -> None:
     """Unlock remote state."""
     remote_state = RemoteStateManager()
     remote_state.unlock()
+
+
+@track(event_name=AnalyticsEvent.PROVISION)
+def provision(
+    location: str,
+    prefix: str,
+    password: str,
+    verbose: Optional[bool] = False,
+) -> MatchaState:
+    """Provision cloud resources using templates.
+
+    Args:
+        location (str): Azure location in which all resources will be provisioned.
+        prefix (str): Prefix used for all resources.
+        password (str): Password for ZenServer.
+        verbose (bool optional): additional output is show when True. Defaults to False.
+
+    Returns:
+        MatchaState: the information of the provisioned resources.
+
+    Raises:
+        MatchaError: If resources are already provisioned.
+        MatchaError: If prefix is not valid.
+        MatchaError: If region is not valid.
+    """
+    remote_state_manager = RemoteStateManager()
+
+    if remote_state_manager.is_state_stale():
```

Comment:
Didn't get anywhere with solving this?

Reply:
If `is_state_stale()` evaluates to True we run `remove_matcha_config()` which sets the `_azure_storage` class variable to `None`. This means that when the property `azure_storage` is requested again it will re-instantiate it with the valid config file relating to the existing deployed resources.